packages/template-retail-react-app/app/components/unavailable-product-confirmation-modal/index.test.js

```diff
@@ -10,349 +10,86 @@ import UnavailableProductConfirmationModal from '@salesforce/retail-react-app/ap
-                            name: 'S',
-                            orderable: true,
-                            value: '9SM'
-                        },
-                        {
-                            name: 'M',
-                            orderable: true,
-                            value: '9MD'
-                        },
-                        {
-                            name: 'L',
-                            orderable: true,
-                            value: '9LG'
-                        },
-                        {
-                            name: 'XL',
-                            orderable: true,
-                            value: '9XL'
-                        }
-                    ]
-                }
-            ],
-            variationValues: {
-                color: 'JJ0NLD0',
-                size: '9MD'
-            },
-            c_color: 'JJ0NLD0',
-            c_refinementColor: 'black',
-            c_size: '9MD',
-            c_width: 'Z'
+            }
+        },
+        {
+            currency: 'GBP',
+            id: '701642889829M',
+            imageGroups: [],
+            inventory: {
+                ats: 68,
+                backorderable: false,
+                id: 'inventory_m',
+                orderable: false,
+                preorderable: false,
+                stockLevel: 0
+            }
         }
     ]
 }
 
 describe('UnavailableProductConfirmationModal', () => {
-    beforeEach(() => {
+    test('renders without crashing', () => {
         prependHandlersToServer([
             {
                 path: '*/products',
                 res: () => {
-                    return mockProducts
+                    return mockProductsWithUnavailableProducts
                 }
             }
         ])
-    })
-    test('renders without crashing', () => {
         expect(() => {
             renderWithProviders(<UnavailableProductConfirmationModal />)
         }).not.toThrow()
     })
 
     test('opens confirmation modal when unavailable products are found', async () => {
-        const mockProductIds = ['701642889899M', '701642889830M']
+        prependHandlersToServer([
+            {
+                path: '*/products',
+                res: () => {
+                    return mockProductsWithUnavailableProducts
+                }
+            }
+        ])
         const mockFunc = jest.fn()
+        const basket = {
+            productItems: [
+                {
+                    productId: '701642889830M',
+                    quantity: 2
+                },
+                // unavailable product
+                {
+                    productId: '701642889829M',
+                    quantity: 3
+                }
+            ]
+        }
         const {getByText, queryByText, queryByRole, user} = renderWithProviders(
             <UnavailableProductConfirmationModal
-                productIds={mockProductIds}
+                productItems={basket.productItems}
```

Comment:
How about adding a test to test the old `productIds` prop? We'll need to make sure that it won't break existing users.